### PR TITLE
bugfix/19816-ordinal-incorrect-zoom

### LINKED ISSUE #19816: Zoom works incorrectly when adding points dynamically and various settings.
#### Expected behaviour

Zoom expands outside current visible range

#### Actual behaviour

Zoom stays within current visible range

#### Live demo with steps to reproduce

Code: https://jsfiddle.net/z72kench/3/

1. Go to the very right edge of the chart
2. Select `1m` (may have to do this multiple times seems like another bug here)
3. Notice the range selector is set to 1 months worth of data
4. Use the zoom feature to select the left part of the chart including the edge
5. See the range has changed to something far outside the original `1m` range

https://github.com/highcharts/highcharts/assets/1490044/bd88128d-75f9-40fe-b391-c2481bebc5d2

#### Product version

- Highstock JS v11.1.0 (2023-06-05)

#### Affected browser(s)

- Windows
  - Chrome
  - Firefox
  - Edge

### PR DESCRIPTION
**Pull request by DJTechnoo**
Fixed #19816, incorrect zooming with multiple series and ordinal axis

### CODE REVIEW

File: samples/unit-tests/axis/ordinal/demo.js [hunk 1]
```diff
@@ -830,4 +830,351 @@ QUnit.test('Moving annotations on ordinal axis.', assert => {
         0.0001,
         'toValue <-> toPixels translation should return the same initial value, #16784. '
     );
-});
\ No newline at end of file
+});
+
+QUnit.test('Zooming in ordnial', assert => {
```
Comment (kamil-musialowski) [inline samples/unit-tests/axis/ordinal/demo.js:835]: Typo in "ordnial" :D

File: samples/unit-tests/axis/ordinal/demo.js [hunk 2]
```diff
@@ -830,4 +830,351 @@ QUnit.test('Moving annotations on ordinal axis.', assert => {
...
+                }
+            }
+        }
+    });
+
+    const testController = new TestController(chart),
+        start = [chart.plotLeft, chart.plotTop + 40],
+        end = [chart.plotLeft + chart.xAxis[0].len / 10, chart.plotTop + 40];
+
+    testController.pan(start, end, {}, true);
```
Comment (kamil-musialowski) [inline samples/unit-tests/axis/ordinal/demo.js:1174]: Unnecessary debug marker enabled

**Review by kamil-musialowski — CHANGES_REQUESTED** [review]
Absolutely minor stuff to be changes, other than that I like the solution and decision about  always using the extended ordinal positions. Good job!

**Review by kamil-musialowski — APPROVED** [review]
That's it, thanks!

**Review by pawellysy — APPROVED** [review]
LGTM

**Comment by TorsteinHonsi** [comment]
Pending resolution to Pawel's comment.

File: ts/Core/Axis/OrdinalAxis.ts [hunk 3]
```diff
@@ -462,31 +462,17 @@ namespace OrdinalAxis {
...
-            isInside = val >= positions[0] &&
-                val <= positions[positions.length - 1];
-
-        // If the value is not inside the plot area, use the extended positions.
-        // (array contains also points that are outside of the plotArea).
-        if (!isInside) {
-            positions = ordinal.getExtendedPositions();
-        }
+        // Always use extendedPositions (#19816)
+        let positions = ordinal.getExtendedPositions();
```
[the remark this answers is not in the document]
Comment (pawelfus) [reply ts/Core/Axis/OrdinalAxis.ts:466]: Debugged together: Postponing this refactor, as it seems to be a longer process than a quick refactor

**Review by pawelfus — APPROVED** [review]
Thank you!

**Review by TorsteinHonsi — APPROVED** [review]

**Pull request merged**